server/service/src/sensor/query.rs

```diff
@@ -37,3 +42,82 @@ pub fn get_sensor(ctx: &ServiceContext, id: String) -> Result<Sensor, SingleReco
         Err(SingleRecordError::NotFound(id))
     }
 }
+
+pub fn get_sensor_logs_for_breach(
+    connection: &StorageConnection,
+    breach_id: &String,
+) -> Result<Vec<TemperatureLog>, RepositoryError> {
+    let mut temperature_logs: Vec<TemperatureLog> = Vec::new();
+
+    let breach_result =
+        TemperatureBreachRowRepository::new(connection).find_one_by_id(breach_id)?;
+
+    if let Some(breach_record) = breach_result {
+        if let Some(end_datetime) = breach_record.end_datetime {
+            // Find all temperature logs in the breach time range, sorted by date/time
+
+            let mut filter = TemperatureLogFilter::new()
+                .sensor(SensorFilter::new().id(EqualFilter::equal_to(&breach_record.sensor_id)));
+            let sort = Sort {
+                key: TemperatureLogSortField::Datetime,
+                desc: None,
+            };
+
+            match breach_record.r#type {
```

Comment:
I'd noticed that as well, but I have a feeling that it may be more complicated - https://github.com/msupply-foundation/open-msupply/issues/2690#issuecomment-1885092383

Reply:
Yeah there is definitely compilation with DLS, but I think that would happen regardless conversion to local time ?

Reply:
It's hard to see any way to nicely handle DST jumps - whether we stick with local time or change to UTC, the timestamps in the Berlinger files will be out by an hour for half the year (until such times as the Berlinger firmware is updated to take account of DST/TZ) => at least if we change to UTC, then it's independent of DST and it's in line with the rest of the DB. 

However... we do store the timestamp of when the sensor is initially activated, so that determines its (fixed) UTC offset => what if we always use that offset to convert to UTC, and then any UI display of the UTC timestamp will take account of DST and show the "real" local time?

Reply:
> However... we do store the timestamp of when the sensor is initially activated

I wonder if we always know that ? Since we will be integrating berlinger data for existing sensors ? Hmm, for some reason i am think DST jump for berlinger data is not that important to capture, not sure why that is i think it's because:
* The frequency of berlinger sensor data integration is pretty low, and the precision of events is not as important as captured the existence of the events
* My previous leaning towards just getting the data into our system, as is, but matching our data modal, vs adjusting the data to be more 'accurate' then what's captured by fridgetag